File: ios/StoreLink/StoreLinkCoreView.swift
//
//  StorelinkCoreView.swift
//  Storelink
//
import SwiftUI

public struct StorelinkCoreView: UIViewControllerRepresentable {
    var refreshToken: String
    var logLevel: LogLevel?
    var viewType: ViewType
    var functionParams: [AnyHashable: Any]?
    var onComplete: (([AnyHashable: Any]) -> Void)?
    var brandName: String?
    var logoUrl: String?
    var _devApiLocation: String?

    public init(
      refreshToken: String,
      logLevel: LogLevel? = nil,
      viewType: ViewType = .backgroundTask,
      functionParams: [AnyHashable: Any]? = nil,
      onComplete: (([AnyHashable: Any]) -> Void)? = nil,
      brandName: String? = nil,
      logoUrl: String? = nil,
      _devApiLocation: String? = nil
    ) {
        self.refreshToken = refreshToken
        self.viewType = viewType
        self.logLevel = logLevel
        self.functionParams = functionParams
        self.onComplete = onComplete
        self.brandName = brandName
        self.logoUrl = logoUrl
        self._devApiLocation = _devApiLocation
    }
  
    public typealias UIViewControllerType = UINavigationController

    public func makeUIViewController(context: Context) -> UINavigationController {
        // Create the SDKViewController with the refreshToken and viewType
      let sdkVC = StorelinkCore.SDKHandler.SDKViewController(
        refreshToken: refreshToken,
        viewType: viewType,
        logLevel: logLevel,
        functionParams: functionParams,
        onComplete: onComplete,
        brandName: brandName,
        logoUrl: logoUrl,
        _devApiLocation: _devApiLocation
      )
        let navigationController = UINavigationController(rootViewController: sdkVC)
        return navigationController
    }

    public func updateUIViewController(_ uiViewController: UINavigationController, context: Context) {
        // Update code if needed
    }
}

File: android/app/src/main/java/com/storelinksdk/StorelinkCoreView.java
//
//  StorelinkCoreView.swift
//  Storelink
//
import android.content.Context;
import android.view.View;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.FragmentActivity;

import java.util.Map;

public class StorelinkCoreView {
    private String refreshToken;
    private LogLevel logLevel;
    private ViewType viewType;
    private Map<String, Object> functionParams;
    private OnCompleteListener onComplete;
    private String brandName;
    private String logoUrl;
    private String devApiLocation;

    public StorelinkCoreView(String refreshToken,
                             @Nullable LogLevel logLevel,
                             ViewType viewType,
                             @Nullable Map<String, Object> functionParams,
                             @Nullable OnCompleteListener onComplete,
                             @Nullable String brandName,
                             @Nullable String logoUrl,
                             @Nullable String devApiLocation) {
        this.refreshToken = refreshToken;
        this.logLevel = logLevel;
        this.viewType = viewType;
        this.functionParams = functionParams;
        this.onComplete = onComplete;
        this.brandName = brandName;
        this.logoUrl = logoUrl;
        this.devApiLocation = devApiLocation;
    }

    public View createView(@NonNull Context context) {
        FragmentActivity activity = (FragmentActivity) context;
        StorelinkCore.SDKHandler.SDKFragment sdkFragment = StorelinkCore.SDKHandler.SDKFragment.newInstance(
                refreshToken,
                viewType,
                logLevel,
                functionParams,
                onComplete,
                brandName,
                logoUrl,
                devApiLocation
        );

        activity.getSupportFragmentManager()
                .beginTransaction()
                .add(android.R.id.content, sdkFragment)
                .commit();

        return new View(context);
    }

    public void updateView(View view) {
        // Update code if needed
    }

    public interface OnCompleteListener {
        void onComplete(Map<String, Object> data);
    }

    public enum ViewType {
        BACKGROUND_TASK,
        TRANSFER_CART
        // Add other view types as needed
    }
}
